### ISSUE TITLE
[backend]: add a logger for all mutations and errors

### ISSUE THREAD

**Issue by nathanschwarz** [Maagan-Michael/net-n#7]
- [ ] add `loguru` as dependency
- [ ] implement a logger for each "repository"
- [ ] use the repository logger for mutations and errors

**Comment by nathanschwarz** [comment]
@yarons when upserting do we need to log the exact changes (fields / new values) ?

**Comment by yarons** [comment]
Yes fully but I think we should make it clear what exactly is happening:
1. A replacement template - `[INFO] {timestamp - {actor}: Family {family} moved from residence {old_residence} to residence {new_residence}.`
2. Assignment - `[INFO] {timestamp} - {actor}: Family {family} was assigned to {new_residence}.`
3. Displacement - `[WARN] {timestamp} - {actor}: Family {family} was displaced.`
4. Failure - `[ERROR] {timestamp} - {actor}: Family {family} was not assigned to {new_residence}, Reason: {reason}.`

First case informs of a successful transaction where the former house was detached, the new house was assigned and the budget is sufficient.
Second case informs of a successful transaction where for some reason this family was previously displaced due to some error (either budget or technical).
Third case is a warning since the user was displaced but not reassigned (if it's a UI decision it's intentional and informational but should popup a warning message saying that it wasn't assigned, if it's done by a script it's a warning) these cases happen either when there's insufficient funds or the user is leaving the Kibbutz.
Fourth case is clearly an error, since it might be the outcome of insufficient funds or technical error (API inaccessible) etc.

The technical errors may include:
1. Inaccessible Ma'alé/Oracle DB (Ingestion).
2. Inaccessible Internal DB/Postgres (Processing/Storage).
3. Inaccessible controller (IMC/SNMP/etc.)

Thanks.

**Comment by nathanschwarz** [comment]
- [ ] replacement log
- [ ] assignment log
- [ ] displacement log
- [ ] failure
